### ISSUE TITLE
[BUG] rerun dump doesn't output more than one frame

### ISSUE THREAD

**Issue by mdelyser** [lammps/lammps#3402]
**Summary**

In "old" versions of lammps (approx Sept 2021 and older), when I use the rerun command with an existing dump file, the newly produced dump file contains data for all frames. In "new" versions of lammps (approx Jan 2022 and newer), when I use the rerun command with an existing dump file, the newly produced dump file only contains data for the first frame (despite still giving me thermodynamic information for each frame). 

**LAMMPS Version and Platform**

I noticed this issue is present in the LAMMPS versions from Jan 7, 2022 and Aug 3, 2022. It is not present in the LAMMPS versions from Dec 12, 2018 and Sept 20, 2021.

**Expected Behavior**

I expected the rerun to produce a dump file with data from every frame as it did with old versions.

**Actual Behavior**

It only gives data from the first frame in the dump file.

**Steps to Reproduce**

Rerun a simulation with a lammps version from this year. Try to output a new dump file for every frame in the existing dump file.

**Further Information, Files, and Links**

[rrbug.tar.gz](https://github.com/lammps/lammps/files/9339922/rrbug.tar.gz)

input_initial.lmp is used to generate the initial simulation. It can be run with any version of lammps. it generates log_initial.lammps, dump_orig.txt, and out.data
input_rerun.lmp was run with a new version of lammps. it generates log_rerun.lammps and dump_pt.txt.
input_rerun_old.lmp was run with an old version of lammps. it generates log_rerun_old.lammps and dump_pt_old.txt.

The only difference between input_rerun.lmp and input_rerun_old.lmp is that i changed the name of the dump file they produce so it doesn't get overwritten when running with different versions. 

dump_pt.txt only contains the data from the first frame, whereas dump_pt_old.txt contains the data from all (both) frames.

**Comment by akohlmey** [comment]
Thanks for your report. Unfortunately (or fortunately?), I cannot reproduce it with either the latest stable version (23 June 2022) or the current development branch (post 3 August 2022). This seems to be a duplicate of issue #3274. There *was* a problem with rerun causing errors or not producing output beyond the first frame and this was due to internal changes that allow to output dumps at regular simulation time intervals (PR #3053). Those issues were addressed in multiple releases over time, the last fix for this particular issue was in PR #3275 released as version 2 June 2022.

**Comment by mdelyser** [comment]
That's weird. I cloned the github repository last Thursday (Aug 11) to make sure the bug was still present. Running the example I sent only gave me 1 frame of output. I just cloned the github repository again, ran the example, and got both frames of output. 

I'm not sure what I messed up last week. Maybe I didn't call the newly compiled version last Thursday when I thought I did. But it's working fine for me now. Sorry about that.
